### PR TITLE
refactor(server): inject database dependency into credential-providers

### LINKED ISSUE #589: Inject database dependency into credential-providers.ts
Parent epic: #577

## Problem

`getCredentialProviders()` and `setCredentialProviders()` in `packages/server/src/credential-providers.ts` import the `db` module directly and call `db.select()` / `db.insert()`. Since `db.ts` runs `initDatabase()` at import time, any test that imports credential-providers also initializes SQLite.

JSON parsing is also embedded in the getter with no separation.

## Proposed Changes

1. Accept an optional `Database` instance parameter (or use the pattern established in #580)
2. Extract JSON parsing into a pure function for independent testing
3. Add unit tests for credential provider logic using a mock database

## Files

- `packages/server/src/credential-providers.ts`
- `packages/server/src/db.ts` (dependency)

### PR DESCRIPTION
**Pull request by nick-pape**
## Summary
- Add optional `database` parameter to `getCredentialProviders()`, `setCredentialProviders()`, and `buildProviderTokenBundle()` so tests can inject an in-memory db directly instead of using `vi.mock("./db.js")`
- Extract `parseCredentialProviderConfig()` as a standalone pure function for JSON parsing/validation, enabling tests with no db dependency at all
- Add 5 new pure-function tests for `parseCredentialProviderConfig()` and remove the `vi.mock("./db.js")` block from the test file

## Test plan
- [x] All 21 credential-providers tests pass (5 new + 16 existing)
- [x] Full server test suite passes (429 tests, 36 files)
- [x] `rush build` compiles cleanly
- [x] No caller changes needed — all existing call sites use the default module-level db

Closes #589

### CODE REVIEW

File: packages/server/src/credential-providers.ts [hunk 1]
```diff
@@ -3,14 +3,15 @@
...
 import { eq } from "drizzle-orm";
+import type { BetterSQLite3Database } from "drizzle-orm/better-sqlite3";
 import { existsSync, readFileSync } from "node:fs";
 import { homedir } from "node:os";
 import { join } from "node:path";
 import { create } from "@bufbuild/protobuf";
 import { powerline, type RuntimeName } from "@grackle-ai/common";
 import db from "./db.js";
+import * as schema from "./schema.js";
 import { settings } from "./schema.js";
```
[the remark this answers is not in the document]
Comment (nick-pape) [reply packages/server/src/credential-providers.ts]: Good point — consolidated to a single `import * as schema` and now use `schema.settings` throughout. Dropped the duplicate `import { settings }` line. Fixed in ec9ac09.

File: packages/server/src/credential-providers.ts [hunk 2]
```diff
@@ -42,28 +43,44 @@ export const VALID_CLAUDE_VALUES: ReadonlySet<string> = new Set(["off", "subscri
...
+ * Throws if the JSON is unparseable — callers decide how to handle the error.
+ */
+export function parseCredentialProviderConfig(rawJson: string): CredentialProviderConfig {
+  const parsed = JSON.parse(rawJson) as Partial<CredentialProviderConfig>;
+  return {
+    claude: VALID_CLAUDE_VALUES.has(parsed.claude ?? "") ? parsed.claude! : DEFAULT_CONFIG.claude,
+    github: VALID_TOGGLE_VALUES.has(parsed.github ?? "") ? parsed.github! : DEFAULT_CONFIG.github,
+    copilot: VALID_TOGGLE_VALUES.has(parsed.copilot ?? "") ? parsed.copilot! : DEFAULT_CONFIG.copilot,
+    codex: VALID_TOGGLE_VALUES.has(parsed.codex ?? "") ? parsed.codex! : DEFAULT_CONFIG.codex,
+  };
```
[the remark this answers is not in the document]
Comment (nick-pape) [reply packages/server/src/credential-providers.ts]: Good catch — now normalizes non-object parsed values (null, numbers, strings, booleans) to `{}` before field validation, so they fall back to defaults instead of throwing. Updated docstring and added a test covering these edge cases. Fixed in 1cbbd9c.
